src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -65,11 +65,18 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
                 }
                 else
                 {
-                    // An EntryAssembly may not be found when running from a custom host.
-                    // Try to find the native entry point.
-                    using (Process currentProcess = Process.GetCurrentProcess())
+                    try
                     {
-                        ApplicationUri = currentProcess.MainModule?.FileName;
+                        // An EntryAssembly may not be found when running from a custom host.
+                        // Try to find the native entry point.
+                        using (Process currentProcess = Process.GetCurrentProcess())
+                        {
+                            ApplicationUri = currentProcess.MainModule?.FileName;
+                        }
+                    }
+                    catch (PlatformNotSupportedException)
```

Comment:
In https://github.com/dotnet/runtime/pull/37944#discussion_r451106998 we said we should do `if (!RuntimeInformation.IsOSPlatform(OSPlatform.Create("BROWSER")))` instead of catching the PNSE.

Reply:
I wanted to keep it platform independent so that we dont have to revisit it for any other new platforms. 

cc @ericstj

Reply:
Right, this code should be tolerant already to not having an entry-point name.  We don't need to bind that decision to a platform.  If anything else comes along that can't provide a process-name that's ok.  

Does anyone else remember the hell of compat shims version detection decisions et-al from years back?  We should be very careful about conflating behavior with platform when making runtime decisions in code that ships in the app (as is the case here: nuget package).

If we care about the linker case (I don't think we do here) we should be designing capability APIs.